### PR TITLE
update to 3.0

### LINKED ISSUE #160: Start and stop not defined in unit tests
I'm trying to implement the unit tests from the code here but getting: 

> 'WebKitBlobBuilder' is not defined.
> 'Blob' is not defined.
> 'start' is not defined.
> 'stop' is not defined.

How can I fix it?

### LINKED ISSUE #167: DEPRECATION: An addon is trying to access project.nodeModulesPath.
I see the following error after `ember s`

DEPRECATION: An addon is trying to access project.nodeModulesPath. This is not a reliable way to discover npm modules. Instead, consider doing: require("resolve").sync(something, { basedir: project.root }). Accessed from:   new NPMDependencyVersionChecker (/Users/username/Projects/datagem-admin-frontend/node_modules/ember-uploader/node_modules/ember-cli-version-checker/src/npm-dependency-version-checker.js:11:33)
DEPRECATION: Overriding init without calling this._super is deprecated. Please call `this._super.init && this._super.init.apply(this, arguments);` addon: `release`

We just need to upgrade `ember-cli-version-checker` to  v2.0.0

### PR DESCRIPTION
**Pull request by snewcomer**
Ready for merge after approval!

close #167 
close #160

### CODE REVIEW

**Comment by Dhaulagiri** [comment]
I re-enabled tests for PRs in travis so the next push to this PR should hopefully trigger a CI run 🤞 

@snewcomer let me know when this is no longer WIP and we can get it :shipit:

**Comment by snewcomer** [comment]
@Dhaulagiri 👋  Got the tests passing and tested this branch out locally and all seems good (although I don't use the full API of this addon).  In addition to upgrading, I ran the qunit codemod and the modules codemod.

File: package.json [hunk 1]
```diff
@@ -12,46 +12,52 @@
...
-    "ember-cli-dependency-checker": "^1.2.0",
-    "ember-cli-eslint": "1.2.1",
-    "ember-cli-htmlbars": "^1.0.3",
-    "ember-cli-htmlbars-inline-precompile": "^0.3.1",
-    "ember-cli-inject-live-reload": "^1.4.0",
-    "ember-cli-mirage": "0.1.11",
-    "ember-cli-qunit": "^1.4.0",
+    "babel-plugin-transform-object-rest-spread": "^6.26.0",
+    "broccoli-asset-rev": "^2.4.5",
+    "ember-ajax": "^3.0.0",
```
Comment (Dhaulagiri) [inline package.json]: I'm fairly certain that this isn't actually used and could be removed

File: package.json [hunk 2]
```diff
@@ -12,46 +12,52 @@
...
-    "node": ">= 0.10.0"
+    "node": "^4.5 || 6.* || >= 7.*"
+  },
+  "dependencies": {
+    "broccoli-file-creator": "^1.0.1",
+    "broccoli-merge-trees": "^1.0.0",
+    "ember-cli-babel": "^6.6.0"
   },
   "devDependencies": {
     "babel-eslint": "^4.1.8",
```
Comment (Dhaulagiri) [inline package.json]: i wonder if this is actually used? seems unusual for an ember project.

File: MODULE_REPORT.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
+## Module Report
```
Comment (Dhaulagiri) [inline MODULE_REPORT.md]: I don't think we want to commit this

File: package.json [hunk 4]
```diff
@@ -12,46 +12,52 @@
...
+    "ember-source": "~3.1.2",
+    "ember-source-channel-url": "^1.1.0",
+    "ember-try": "^0.2.23",
+    "eslint-plugin-ember": "^5.2.0",
+    "eslint-plugin-node": "^6.0.1",
     "glob": "^4.5.3",
     "js-string-escape": "^1.0.0",
-    "loader.js": "^4.0.1",
+    "loader.js": "^4.2.3",
     "morgan": "^1.6.1",
```
Comment (Dhaulagiri) [inline package.json]: i'm not convinced this dep is actually being used

File: package.json [hunk 5]
```diff
@@ -12,46 +12,52 @@
...
+    "ember-resolver": "^4.0.0",
+    "ember-sinon": "2.1.0",
+    "ember-sinon-qunit": "3.1.0",
+    "ember-source": "~3.1.2",
+    "ember-source-channel-url": "^1.1.0",
+    "ember-try": "^0.2.23",
+    "eslint-plugin-ember": "^5.2.0",
+    "eslint-plugin-node": "^6.0.1",
     "glob": "^4.5.3",
     "js-string-escape": "^1.0.0",
```
Comment (Dhaulagiri) [inline package.json]: this can be removed now

File: package.json [hunk 6]
```diff
@@ -12,46 +12,52 @@
...
+    "ember-maybe-import-regenerator": "^0.1.6",
+    "ember-resolver": "^4.0.0",
+    "ember-sinon": "2.1.0",
+    "ember-sinon-qunit": "3.1.0",
+    "ember-source": "~3.1.2",
+    "ember-source-channel-url": "^1.1.0",
+    "ember-try": "^0.2.23",
+    "eslint-plugin-ember": "^5.2.0",
+    "eslint-plugin-node": "^6.0.1",
     "glob": "^4.5.3",
```
Comment (Dhaulagiri) [inline package.json]: pretty sure this isn't used either

**Comment by Dhaulagiri** [comment]
Also, if we don't do it here we should make a PR to update the readme to reflect the removal of bower and the addition of yarn

**Comment by snewcomer** [comment]
@Dhaulagiri Really appreciate the thorough review.  Fixed based on your feedback!  Do you have write access?  Seems I don't.

**Comment by Dhaulagiri** [comment]
I do. I'm not sure that I have npm publish access though but in either case I'll cut a new release in #173

**Comment by Dhaulagiri** [comment]
Nope, I don't have access but nevertheless the tagged version could be used via github for now

**Comment by digitaltoad** [comment]
What is your NPM user so I can give you publish rights?

**Comment by snewcomer** [comment]
snewcomer 🙏

**Comment by Dhaulagiri** [comment]
dhaulagiri

**Comment by digitaltoad** [comment]
Alright I added you both. Thank you again. If you want or need to change the location of the repo, just let me know.

**Pull request merged**